flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -107,6 +112,17 @@ protected ShardConsumer(KinesisDataFetcher<T> fetcherRef,
 		this.fetchIntervalMillis = Long.valueOf(consumerConfig.getProperty(
 			ConsumerConfigConstants.SHARD_GETRECORDS_INTERVAL_MILLIS,
 			Long.toString(ConsumerConfigConstants.DEFAULT_SHARD_GETRECORDS_INTERVAL_MILLIS)));
+
+		if (lastSequenceNum.equals(SentinelSequenceNumber.SENTINEL_AT_TIMESTAMP_SEQUENCE_NUM.get())) {
```

Comment:
I think it might be cleaner if we extract / parse the `initTimestamp` in the sentinel sequence number case determination at the beginning of run().

As for checking if the timestamp is parseable, we should do that when validating the user-provided property configs locally at the job client. You can take a look at the constructor code of `FlinkKinesisConsumer` - that's where we validate the properties. This validation is done at the job client, so that if in any case the properties is errorneous / unparseable, we can handle that _before_ a Flink job is actually launched.